core/modules/layout/plugins/context/layout_context.inc

```diff
@@ -257,6 +259,9 @@ abstract class LayoutContext extends LayoutHandler {
           $this->setData($context_data);
         }
       }
+
+      // Set the raw argument in the context.
+      $this->argument = $router_original_map[$this->position];
```

Comment:
Its extremely unlikely, but contrib could be calling `setDataFromRouterMap()` without this new parameter, so we should wrap this in conditions. 
```
if (isset($router_original_map)) {
      $this->argument = $router_original_map[$this->position];
}
```